crates/runtime/src/datafusion.rs

```diff
@@ -1334,6 +1340,27 @@ impl DataFusion {
             }
         }
     }
+
+    /// Create or get a logical plan from the query
+    async fn get_or_create_logical_plan(
```

Comment:
We must already have a function to do this from results cache for `cache_key_type: plan`? Can we not just use that almost directly?

Reply:
It looks like this

```rust
    async fn try_get_cached_result(
        df: &DataFusion,
        request_context: Arc<RequestContext>,
        mut tracker: QueryTracker,
        key: CacheKey<'_>,
    ) -> super::Result<CacheResult>
```

I believe it requires a bit of refactor if we want to make that generic for getting cached results and getting cached plans. Not sure it's worth it versus just adding ~20LOC.